spec/requests/create_bookmark_from_email_spec.rb

```diff
@@ -13,12 +13,23 @@
     expect(@user.topics.first.bookmarks).to match_array [Bookmark.find_by(url: "www.google.com")]
     expect(response.status).to eq(200) 
   end
-  xit "creates a bookmark if topic already exists" do
+  it "creates a bookmark if topic already exists" do
+    topic = create(:topic, title: "animals", user: @user)
+    post '/bookmarks',{sender: "test@example.com", subject: "animals", "body-plain": "www.google.com"}
+    expect(@user.topics.first.bookmarks.first).to belong_to(:topic)
   end
-  xit "does not create a topic nor a bookmark if user does not exists" do 
+  it "does not create a topic nor a bookmark if user does not exists", :skip_before do
+    post '/bookmarks',{sender: "test@example.com", subject: "animals", "body-plain": "www.google.com"}
+    expect(Topic.all.count).to eq(0)
+    expect(Bookmark.all.count).to eq(0)
+    expect(response.status).to eq(200) 
   end
-  xit "does not create a bookmark without a topic" do 
+  it "does not create a bookmark without a topic" do 
+    expect{post '/bookmarks',{sender: "test@example.com", subject: "", "body-plain": "www.google.com"}
+       }.to raise_error(ActiveRecord::RecordInvalid)
```

Comment:
Reformat this `it` block and the `it` block below to read a bit clearer:

```ruby
expect do
  post '/bookmarks', { PARAMS GO HERE }
end.to raise_error(ActiveRecord::RecordInvalid)
```

In addition, add a check to the end of each `it` block that no bookmarks and/or topics have been created. Currently the tests would pass if the bookmark was created before the error is raised by the controller.